engine/src/core/core.cpp

```diff
@@ -136,6 +140,37 @@ int run(Engine* const engine) try
     auto previousTime = std::chrono::high_resolution_clock::now();
     auto startTime = std::chrono::steady_clock::now();
     // TODO: consider if we should provide an option to render firstly to the window then copy it to the headset
+
+#ifdef CYBSDK_FOUND
+    CybSDK::VirtDevice* device = CybSDK::Virt::FindDevice();
+    if (device == nullptr)
+    {
+        LOGGER_ERR("Cyberith Virtualizer device not found");
+    }
+
+    const CybSDK::VirtDeviceInfo& info = device->GetDeviceInfo();
+    const wchar_t* virtualizerName = info.ProductName;
+    size_t virtualizerNameLen = 0;
+    if (wcstombs_s(&virtualizerNameLen, nullptr, 0, virtualizerName, 0) == 0)
+    {
+        char* mbstr = new char[virtualizerNameLen];
+        wcstombs_s(&virtualizerNameLen, mbstr, virtualizerNameLen, virtualizerName, virtualizerNameLen);
+        std::string virtualizerConvertedName(mbstr);
```

Comment:
this solution is overcomplicated and in c style
you can just calculate productName c string size then create std::string with the previously gained size to finally use wcstombs and print the filled std string